File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i, value in enumerate(nums):
            remaining = target - nums[i]
           
            if remaining in seen:
                return [i, seen[remaining]]
            seen[value] = i 

File: 1-two-sum/1-two-sum.rs
pub mod Solution {
    use std::collections::HashMap;

    pub fn two_sum(nums: Vec<usize>, target: usize) -> Vec<usize> {
        let mut memo: HashMap<usize, usize> = HashMap::new();
        let mut result = Vec::new();
        
        for (index, value) in nums.iter().enumerate() {
            let remain = target - nums[index];
            
            if memo.contains_key(&remain) {
                result = vec![index, memo[&remain]];
                break;
            }
            memo.insert(*value, index);
        }
        result
    }
}
